java_client/ChitChatDesktop/src/com/moc/chitchat/controller/authentication/UserSearchController.java

```diff
@@ -0,0 +1,63 @@
+package com.moc.chitchat.controller.authentication;
+
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.JsonNode;
+import com.mashape.unirest.http.exceptions.UnirestException;
+import com.moc.chitchat.client.HttpClient;
+import com.moc.chitchat.exception.UnexpectedResponseException;
+import com.moc.chitchat.model.UserModel;
+import com.moc.chitchat.resolver.UserResolver;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * UserSearchController is the controller for the user search feature
+ */
+@Component
+public class UserSearchController {
+
+    private HttpClient httpClient;
+    private UserResolver userResolver;
+
+    @Autowired
+    public UserSearchController(
+            HttpClient httpClient,
+            UserResolver userResolver
+    ){
+        this.httpClient = httpClient;
+        this.userResolver = userResolver;
+    }
+
+    public List<UserModel> searchUser(String username) throws UnirestException, UnexpectedResponseException {
+        Map<String, Object> mapper = new HashMap<>();
+        mapper.put("username", username);
+
+        HttpResponse<JsonNode> response = this.httpClient.get("/api/v1/users", mapper);
+
+        if (response.getStatus() != 200) {
```

Comment:
In fact, take a look at this (from Login/Register):

```
if (response.getStatus() == 422) {
            this.userValidator.throwErrorsFromResponse(response);
```

When there's less than 3 characters, the backend performs validation and returns `400` with this body:
```
{
  "errors": {
    "username": [
      "should be at least 3 character(s)"
    ]
  }
}
```
which you can use with `this.userValidator.throwErrorsFromResponse(response);`. And change the condition to `400` instead of `422`. Then you can use the response in the View too :)

Sorry, i'd forgotten to update the trello card, i've done that now: https://trello.com/c/o4Ux4gIR/5-user-can-search-users-list.